src/logic.js

```diff
@@ -23,23 +23,42 @@ function near(candidateRect, refRect, offset = 30) {
   );
 }
 
-function distance(candidateBoundingBox, refBoundingBox) {
-  let leftDiff = Math.abs(candidateBoundingBox.x - refBoundingBox.x);
-  let rightDiff = Math.abs(
-    candidateBoundingBox.x +
-      candidateBoundingBox.width -
-      (refBoundingBox.x + refBoundingBox.width)
+function distance(candidateBoundingBox, refBoundingBox, proximitySelector) {
```

Comment:
I still have to read this function carefully. Sorry, I literally only have 10 minutes here and there

Reply:
No worry, first look tests that I have added.
Then I think I should discut about that: 
```
if (
    proximitySelector === "toLeftOf" ||
    proximitySelector === "toRightOf"
  ) {
    yDiff = yDiff * 10; // Give priority on distance X
  }
```
It to give a priority on the requested direction. When you find a composant at Left you expect that the components must be really at Left.
If we have a component at 20px left and top 0.
And a second at 2px left but 15px top.
That trick force to select the first one.

The value 10 look good but should be precise with real use case.

What do you think?

Reply:
Feels a bit like cheating, but it's better than the current implementation

Reply:
@jfgreffier I found a case where the distance was not correct. I add a new test to covert it "should not far if bounded" and fix the issue.

Reply:
> Feels a bit like cheating, but it's better than the current implementation

Finnaly I remove this 'priority trick', I thinks it's better to it keep simple and improve with real case in second step.